go/vt/vtgate/planbuilder/abstract/queryprojection.go

```diff
@@ -61,33 +62,42 @@ type (
 	}
 )
 
+// GetExpr returns the underlying sqlparser.Expr of our SelectExpr
+func (s SelectExpr) GetExpr() (sqlparser.Expr, error) {
+	switch sel := s.Col.(type) {
+	case *sqlparser.StarExpr:
+		return nil, vterrors.New(vtrpcpb.Code_INTERNAL, "starExpr does not have expr")
+	case *sqlparser.AliasedExpr:
+		return sel.Expr, nil
+	case *sqlparser.Nextval:
+		return sel.Expr, nil
+	}
+	return nil, nil
+}
+
+// GetAliasedExpr returns the SelectExpr as a *sqlparser.AliasedExpr if its type allows it,
+// otherwise an error is returned.
+func (s SelectExpr) GetAliasedExpr() (*sqlparser.AliasedExpr, error) {
+	switch expr := s.Col.(type) {
+	case *sqlparser.AliasedExpr:
+		return expr, nil
+	case *sqlparser.StarExpr:
+		return nil, vterrors.Errorf(vtrpcpb.Code_UNIMPLEMENTED, "unsupported: '*' expression in cross-shard query")
```

Comment:
This seems wrong, we should probably return a global error and catch it from the outside and decide whether we want to return an `expression in cross-shard query` error, like we did for multiple tables predicates